script.js

```diff
@@ -203,3 +203,25 @@ function validateForm(event) {
 }
 
 onclick.addEventListener('submit', validateForm);
+
+// start-local-storage-functionality
+const form = document.getElementById('contact-form');
+const input = document.getElementById('name');
+const userEmail = document.getElementById('email');
+const text = document.getElementById('message');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  localStorage.setItem('myInputValue', input.value);
+  localStorage.setItem('myEmailValue', userEmail.value);
+  localStorage.setItem('myTextValue', text.value);
+
+  form.reset();
+});
+
+if (localStorage.getItem('myInputValue')) {
+  input.value = localStorage.getItem('myInputValue');
+  userEmail.value = localStorage.getItem('myEmailValue');
+  text.value = localStorage.getItem('myTextValue');
+}
```

Comment:
- [ ] Nice work so far 👍🏿  but I notice that when I filled the form and submit my data are not saved to the local storage and on reloads the form inputs are empty. I kindly suggest you fix these issues.
 - something like this 
 ```
const dataStorage = {};
let toStore;
const fullName = document.getElementById('fullname');
const textArea = document.getElementById('textarea');
form.addEventListener('input', () => {
  dataStorage.fullName = fullName.value;
  dataStorage.mail = email.value;
  dataStorage.textArea = textArea.value;
  toStore = JSON.stringify(dataStorage);
  localStorage.setItem('formData', toStore);
});
// retriving
window.addEventListener('DOMContentLoaded', () => {
  const dataRetrieve = JSON.parse(localStorage.getItem('formData'));
  document.getElementById('fullname').value = dataRetrieve.fullName;
  document.getElementById('user-email').value = dataRetrieve.mail;
  document.getElementById('textarea').value = dataRetrieve.textArea;
  errorMessage.textContent = (validation(email.value));
  if (errorMessage.textContent === 'The mail must be writen in lower case') {
    errorMessage.classList.add('error-active');
  } else {
    errorMessage.classList.remove('error-active');
  }
});
```